pgxn/neon/communicator.c

```diff
@@ -665,6 +688,9 @@ consume_prefetch_responses(void)
 {
 	if (MyPState->ring_receive < MyPState->ring_unused)
 		prefetch_wait_for(MyPState->ring_unused - 1);
+
+	MIN_BACKEND_PREFETCH_LSN = InvalidXLogRecPtr; /* No more in-flight prefetch requests from this backend */
+
```

Comment:
It is incorrect to assume that there are no in-flight requests; we've only consumed all prefetch-related requests.

Reply:
... only now. yes.

But it is not wrong for `page_server_request` or `communicator_read_slru_segment` to send the request before it calls `consume_prefetch_requests`, and in either of those cases the lsn of those requests has already been decided on, so _not_ tracking that LSN is definitely incorrect.